### PR TITLE
python310Packages.gb-io: init at 0.1.0

### PR DESCRIPTION
**Pull request by dlesl**
###### Description of changes

https://pypi.org/project/gb-io/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/gb-io/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  };
+
+  sourceRoot = "source";
+
+  nativeBuildInputs = [ setuptools-rust ] ++ (with rustPlatform; [
+    cargoSetupHook
+    rust.cargo
+    rust.rustc
+  ]);
+}
```
Comment (IvarWithoutBones) [inline pkgs/development/python-modules/gb-io/default.nix:31]: You should add the  `meta` attribute

File: pkgs/development/python-modules/gb-io/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  src = fetchFromGitHub {
+    owner = "althonos";
+    repo = "gb-io.py";
+    rev = "v${version}";
+    sha256 = "0s0xm9x8lbcs7pw4bgxlqqhisgrs6g2a37bqkg91hjcz1qis1hl2";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src sourceRoot;
+    name = "${pname}-${version}";
```
Comment (IvarWithoutBones) [inline pkgs/development/python-modules/gb-io/default.nix:20]: Bit nicer to read i think:
```suggestion
    name = "gb-io-${version}";
```

**Comment by IvarWithoutBones** [comment]
Thanks for the PR! I've left a few suggestions :blush:

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
The commit must be `python310Packages.gb-io: init at 0.1.0`.

File: pkgs/development/python-modules/gb-io/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  nativeBuildInputs = [ setuptools-rust ] ++ (with rustPlatform; [
+    cargoSetupHook
+    rust.cargo
+    rust.rustc
+  ]);
+
+  meta = with lib; {
+    homepage = "https://github.com/althonos/gb-io.py";
+    description = "A Python interface to gb-io, a fast GenBank parser written in Rust";
+    platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/gb-io/default.nix:36]:
```suggestion
```

that is the default

File: pkgs/development/python-modules/gb-io/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  };
+
+  sourceRoot = "source";
+
+  nativeBuildInputs = [ setuptools-rust ] ++ (with rustPlatform; [
+    cargoSetupHook
+    rust.cargo
+    rust.rustc
+  ]);
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/gb-io/default.nix:32]:
```suggestion

  checkPhase = ''
    python -m unittest discover
  '';

  pythonImportsCheck = [ "gb_io" ];

```

see https://github.com/althonos/gb-io.py/blob/main/.github/workflows/test.yml

**Pull request merged**